plugins/modules/files/sapcar_extract.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2021, Rainer Leber <rainerleber@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: sapcar_extract
+short_description: Manages SAPCAR from SAP
+version_added: "3.2.0"
+description:
+    - Provides support for unpacking sar/car files with the SAPCAR binary from SAP and pulling
+      information back into Ansible.
+options:
+  path:
+    description: The path to the SAR/CAR file.
+    type: path
+    required: true
+  dest:
+    description:
+      - The destination where SAPCAR extracts the SAR file. Missing folders will be created.
+        If this parameter is not provided it will unpack in the same folder as the SAR file.
+    type: path
+  binary_path:
+    description:
+      - The path to the SAPCAR binary e.g. /home/dummy/sapcar or https://myserver/SAPCAR.
+        If this parameter is not provided the module will try excute from application binary e.g /usr/local/bin.
+    type: path
+  signature:
+    description:
+      - If true the Signature will be extracted
+    default: false
+    type: bool
+  manifest:
+    description:
+      - The Name of the manifest defaults to SIGNATURE.SMF.
+    default: "SIGNATURE.SMF"
+    type: str
+  remove:
+    description:
+      - If true the SAR/CAR file will be removed - !!!This should be used with caution!!!.
+    default: false
+    type: bool
+author:
+    - Rainer Leber (@RainerLeber)
+'''
+
+EXAMPLES = """
+# Extract SAR file - simple
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+    register: sapcar
+
+# Extract SAR file with destination
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      dest: "~/test/"
+    register: sapcar
+
+# Extract SAR file with destination and download from webserver can be a fileshare as well
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      dest: "~/dest/"
+      binary_path: "https://myserver/SAPCAR"
+    register: sapcar
+
+# Extract SAR file and delete SAR after extract
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      remove: true
+    register: sapcar
+
+# Extract SAR file with manifest
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      signature: true
+    register: sapcar
+
+# Extract SAR file with manifest and rename it
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      manifest: "MyNewSignature.SMF"
+      signature: true
+    register: sapcar
+
+"""
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+
+
+def getListOfFiles(dirName):
```

Comment:
Please use Python naming conventions.
```suggestion
def get_list_of_files(dir_name):
```